spec/system/log_in_system_spec.rb

```diff
@@ -17,8 +17,35 @@
       fill_in "user_email", with: "deactivated@example.com"
       fill_in "user_password", with: "password"
       find('input[name="commit"]').click
-
       expect(page).to have_content("Invalid Email or password")
     end
   end
-end
+
+  example "Check for modal in login page for staging env" do
```

Comment:
@nataliagalan this is good! Here is an option on how you can reduce the duplication using some interpolation:
```
  describe 'Showing the modal warning in staging' do
    ["/users/sign_in", "/users/password/new"].each do |path|
      context "when accessing #{path} in the staging environment" do
        before do
          allow(Rails.env).to receive(:staging?).and_return(true)
          visit path
        end

        it 'should render the modal' do
          expect(page).to have_content 'This site is for TEST purposes only!'
        end
      end

      context "when accessing #{path} not in the staging environment" do
        before do
          allow(Rails.env).to receive(:staging?).and_return(false)
          visit path
        end

        it 'should not render the modal' do
          page.assert_no_selector('#warningModal')
        end
      end
    end
  end
```

I used a `.each` loop to check each of the routes. This is beneficial because you don't have to repeat code :)

Reply:
this is beautiful! thank you so much!